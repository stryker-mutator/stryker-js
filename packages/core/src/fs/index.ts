export * from './file-system.js';
export * from './project.js';
export * from './project-file.js';
export * from './project-reader.js';
