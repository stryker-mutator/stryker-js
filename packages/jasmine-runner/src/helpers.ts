import { TestResult, TestStatus } from '@stryker-mutator/api/test-runner';
import jasmine from 'jasmine';

export const helpers = {
  toStrykerTestResult(specResult: jasmine.SpecResult, timeSpentMs: number): TestResult {
    const baseResult = {
      id: specResult.id.toString(),
      name: specResult.fullName,
      timeSpentMs,
    };
    if (specResult.status === 'disabled' || specResult.status === 'pending' || specResult.status === 'excluded') {
      return {
        ...baseResult,
        status: TestStatus.Skipped,
      };
    } else if (!specResult.failedExpectations || specResult.failedExpectations.length === 0) {
      return {
        ...baseResult,
        status: TestStatus.Success,
      };
    } else {
      return {
        ...baseResult,
        status: TestStatus.Failed,
        failureMessage: specResult.failedExpectations.map((failedExpectation) => failedExpectation.message).join(','),
      };
    }
  },
  createJasmine(options: jasmine.JasmineOptions): jasmine {
    return new jasmine(options);
  },
};
