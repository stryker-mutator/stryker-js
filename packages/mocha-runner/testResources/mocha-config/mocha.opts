--async-only
--blaat true
--spec /foo/*.js
/tests/**/*.js