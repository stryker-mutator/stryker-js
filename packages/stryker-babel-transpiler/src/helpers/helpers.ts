import * as path from 'path';

export function toJSFileName(fileName: string) {
  const fileNameInfo = path.parse(fileName);
  const newFileName = path.join(fileNameInfo.dir, fileNameInfo.name + '.js');
  return newFileName;
}
