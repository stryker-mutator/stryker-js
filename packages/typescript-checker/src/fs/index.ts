export * from './hybrid-file-system';
