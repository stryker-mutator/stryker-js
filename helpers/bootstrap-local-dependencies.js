// @ts-check
import { LocalInstaller, progress } from 'install-local';
import fs from 'fs';
import glob from 'glob';
import path from 'path';

console.log('starting installation of local dependencies');

/**
 * @param {string} pattern 
 * @param {glob.IOptions} options 
 * @returns {Promise<string[]>}
 */
function globAsPromised(pattern, options) {
  return new Promise((res, rej) => {
    glob(pattern, options, (err, matches) => {
      if (err) {
        rej(err);
      } else {
        res(matches);
      }
    });
  });
}

/**
 * @typedef Package
 * @property {{ [name: string]: string }} localDependencies

/**
 * Installs local dependencies in one go,
 * reads the package.json files and installs the local dependencies marked there.
 * The packageJsonGlobs default to ./package.json and test/* /package.json files 
 * @param {string} directory
 * @param {string[]} packageJsonGlobs
 * @returns {Promise<void>}
 */
export async function bootstrapLocalDependencies(directory, packageJsonGlobs = ['package.json', 'test/*/package.json']) {
  console.log(`bootstrap ${path.resolve(directory)} (using ${packageJsonGlobs.join(',')})`);
  const files = (await Promise.all(packageJsonGlobs.map((globPattern) => globAsPromised(globPattern, { cwd: path.resolve(directory) })))).flat();
  const packages = await Promise.all(files.map(fileName => fs.promises.readFile(fileName, 'utf-8')
    .then((rawContent) => {
      /**
       * @type {Package}
       */
      const content = JSON.parse(rawContent);
      return { dir: path.dirname(fileName), content };
    })));
  /**
   * @type {import('install-local').ListByPackage}
   */
  const sourcesByTarget = {};
  for (const pkg of packages) {
    const localDeps = pkg.content.localDependencies;
    if (localDeps) {
      sourcesByTarget[path.resolve(pkg.dir)] = Object.keys(localDeps).map(key => path.resolve(pkg.dir, localDeps[key]));
    }
  }
  const localInstaller = new LocalInstaller(sourcesByTarget);
  progress(localInstaller);
  await localInstaller.install();
}
