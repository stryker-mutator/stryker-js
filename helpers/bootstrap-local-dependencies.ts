import { ListByPackage, LocalInstaller, progress } from 'install-local';
import * as fs from 'fs';
import glob = require('glob');
import path = require('path');

console.log('starting installation of local dependencies');

function globAsPromised(pattern: string, options: glob.IOptions) {
  return new Promise<string[]>((res, rej) => {
    glob(pattern, options, (err, matches) => {
      if (err) {
        rej(err);
      } else {
        res(matches);
      }
    });
  });
}

function readFile(fileName: string) {
  return new Promise<string>((res, rej) => fs.readFile(fileName, 'utf8', (err, content) => {
    if (err) {
      rej(err);
    } else {
      res(content);
    }
  }));
}

interface Package {
  localDependencies?: { [name: string]: string };
}

/**
 * Installs local dependencies in one go,
 * reads package.json and test/* /package.json files and installs the local dependencies marked there
 * @param directory the directory where the tests live
 */
export async function bootstrapLocalDependencies(directory: string) {
  console.log('bootstrap ' + path.resolve(directory));
  const files = await globAsPromised('{package.json,test/*/package.json}', { cwd: path.resolve(directory) });
  const packages = await Promise.all(files.map(fileName => readFile(fileName)
    .then(content => ({ dir: path.dirname(fileName), content: JSON.parse(content) as Package }))));
  const sourcesByTarget: ListByPackage = {};
  for (const pkg of packages) {
    const localDeps = pkg.content.localDependencies;
    if (localDeps) {
      sourcesByTarget[path.resolve(pkg.dir)] = Object.keys(localDeps).map(key => path.resolve(pkg.dir, localDeps[key]));
    }
  }
  const localInstaller = new LocalInstaller(sourcesByTarget);
  progress(localInstaller);
  await localInstaller.install();
}
