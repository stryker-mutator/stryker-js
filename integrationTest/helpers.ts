import * as path from 'path';
import * as fs from 'mz/fs';
import { expect } from 'chai';
import { ScoreResult } from 'stryker-api/report';

export async function readScoreResult(eventResultDirectory = path.resolve('reports', 'mutation', 'events')) {
    const allReportFiles = await fs.readdir(eventResultDirectory);
    const scoreResultReportFile = allReportFiles.find(file => !!file.match(/.*onScoreCalculated.*/));
    expect(scoreResultReportFile).ok;
    const scoreResultContent = await fs.readFile(path.resolve(eventResultDirectory, scoreResultReportFile || ''), 'utf8');
    return JSON.parse(scoreResultContent) as ScoreResult;
}