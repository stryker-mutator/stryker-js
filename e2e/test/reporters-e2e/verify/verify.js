import fs from 'fs';

import { expect } from 'chai';
import { describe } from 'mocha';

import { readMutationTestingJsonResult } from '../../../helpers.js';

describe('Verify stryker has ran correctly', () => {
  /**
   * @param {string} fileName
   */
  function expectExists(fileName) {
    expect(fs.existsSync(fileName), `Missing ${fileName}!`).true;
  }

  it('should report the html file', () => {
    expectExists('reports/mutation/mutation.html');
  });

  it('should have a clear text report', () => {
    expectExists('reports/stdout.txt');
  });

  it('should have a json report', () => {
    expectExists('reports/mutation/mutation.json');
  });

  it('should report json report with expected results', async () => {
    const result = await readMutationTestingJsonResult();
    const files = Object.fromEntries(
      Object.entries(result.files)
        .map(([fileName, fileResult]) => [fileName, { ...fileResult, mutants: fileResult.mutants.sort((a, b) => (a.id < b.id ? -1 : 1)) }])
        .sort(([a], [b]) => (a < b ? -1 : 1))
    );
    const testFiles = Object.fromEntries(
      Object.entries(result.testFiles)
        .map(([fileName, fileResult]) => [fileName, { ...fileResult, tests: fileResult.tests.sort((a, b) => (a.id < b.id ? -1 : 1)) }])
        .sort(([a], [b]) => (a < b ? -1 : 1))
    );
    expect(files).matchSnapshot();
    expect(testFiles).matchSnapshot();
  });

  describe('clearText report', () => {
    /**
     * @type {string}
     */
    let stdout;
    beforeEach(async () => {
      stdout = await fs.promises.readFile('reports/stdout.txt', 'utf8');
    });
    it('should report all tests', () => {
      expect(stdout).matches(createTestsRegex());
    });
    it('should report NoCoverage mutants', () => {
      expect(stdout).matches(createNoCoverageMutantRegex());
    });
    it('should report Survived mutants', () => {
      expect(stdout).matches(createSurvivedMutantRegex());
    });
    it('should report average tests per mutant', () => {
      expect(stdout).contains('Ran 0.80 tests per mutant on average.');
    });
    it('should report the clearText table', () => {
      const clearTextTableRegex = createClearTextTableSummaryRowRegex();
      expect(stdout).matches(clearTextTableRegex);
    });
    it('should finish up with the clear text report', () => {
      const clearTextTableRegex = createClearTextTableSummaryRowRegex();
      const survivedMutantRegex = createSurvivedMutantRegex();
      const indexOfSurvivedMutant = survivedMutantRegex.exec(stdout).index;
      const indexOfClearTextTable = clearTextTableRegex.exec(stdout).index;
      expect(indexOfSurvivedMutant).lessThan(indexOfClearTextTable);
    });
  });
});
const createTestsRegex = () => /All tests\s*AddSpec\.js\s*\s*âœ“ Add should be able to add two numbers \(killed 2\)/;
const createNoCoverageMutantRegex = () => /\[NoCoverage\]/;
const createSurvivedMutantRegex = () => /\[Survived\]/;
const createClearTextTableSummaryRowRegex = () => /All files\s*\|\s*64\.00\s*\|\s*16\s*\|\s*0\s*\|\s*1\s*\|\s*8\s*\|\s*0\s*\|/;
