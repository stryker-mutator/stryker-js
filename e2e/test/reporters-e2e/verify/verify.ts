import { expect } from 'chai';
import fs from 'fs';
import { describe } from 'mocha';

describe('Verify stryker has ran correctly', () => {

  function expectExists(fileName: string) {
    expect(fs.existsSync(fileName), `Missing ${fileName}!`).true;
  }

  it('should report html files', () => {
    expectExists('reports/mutation/html/index.html');
    expectExists('reports/mutation/html/mutation-test-elements.js');
    expectExists('reports/mutation/html/stryker-80x80.png');
    expectExists('reports/mutation/html/bind-mutation-test-report.js');
  });

  it('should have a clear text report', () => {
    expectExists('reports/stdout.txt');
  });

  it('should have a json report', () => {
    expectExists('reports/mutation/mutation.json');
  });

  describe('clearText report', () => {

    let stdout: string;
    beforeEach(async () => {
      stdout = await fs.promises.readFile('reports/stdout.txt', 'utf8');
    })

    it('should report NoCoverage mutants', () => {
      expect(stdout).matches(createNoCoverageMutantRegex());
    });

    it('should report Survived mutants', () => {
      expect(stdout).matches(createSurvivedMutantRegex());
    });

    it('should report average tests per mutant', () => {
      expect(stdout).contains('Ran 0.80 tests per mutant on average.');
    });

    it('should report the clearText table', () => {
      const clearTextTableRegex = createClearTextTableSummaryRowRegex();
      expect(stdout).matches(clearTextTableRegex);
    });

    it('should finish up with the clear text report', () => {
      const clearTextTableRegex = createClearTextTableSummaryRowRegex();
      const survivedMutantRegex = createSurvivedMutantRegex();
      const indexOfSurvivedMutant = survivedMutantRegex.exec(stdout).index;
      const indexOfClearTextTable = clearTextTableRegex.exec(stdout).index;
      expect(indexOfSurvivedMutant).lessThan(indexOfClearTextTable);
    });
  })

  
});

const createNoCoverageMutantRegex = () => /#6\.\s*\[NoCoverage\]/;

const createSurvivedMutantRegex = () => /#20\.\s*\[Survived\]/;

const createClearTextTableSummaryRowRegex = () => /All files\s*\|\s*64\.00\s*\|\s*16\s*\|\s*0\s*\|\s*1\s*\|\s*8\s*\|\s*0\s*\|/;
