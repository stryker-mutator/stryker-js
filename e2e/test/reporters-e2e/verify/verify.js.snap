// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify stryker has ran correctly should report json report with expected results 1`] = `
Object {
  "src/Add.js": Object {
    "language": "javascript",
    "mutants": Array [
      Object {
        "coveredBy": Array [
          "0",
        ],
        "id": "0",
        "killedBy": Array [
          "0",
        ],
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 3,
          },
          "start": Object {
            "column": 43,
            "line": 1,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected undefined to equal 7",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "0",
        ],
        "id": "1",
        "killedBy": Array [
          "0",
        ],
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 2,
          },
          "start": Object {
            "column": 10,
            "line": 2,
          },
        },
        "mutatorName": "ArithmeticOperator",
        "replacement": "num1 - num2",
        "static": false,
        "status": "Killed",
        "statusReason": "expected -3 to equal 7",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [],
        "id": "10",
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 15,
          },
          "start": Object {
            "column": 10,
            "line": 15,
          },
        },
        "mutatorName": "EqualityOperator",
        "replacement": "number <= 10",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "11",
        "killedBy": Array [
          "3",
        ],
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 24,
          },
          "start": Object {
            "column": 52,
            "line": 18,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected undefined to be true",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "12",
        "killedBy": Array [
          "4",
        ],
        "location": Object {
          "end": Object {
            "column": 25,
            "line": 19,
          },
          "start": Object {
            "column": 20,
            "line": 19,
          },
        },
        "mutatorName": "BooleanLiteral",
        "replacement": "true",
        "static": false,
        "status": "Killed",
        "statusReason": "expected true to be false",
        "testsCompleted": 2,
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "13",
        "killedBy": Array [
          "4",
        ],
        "location": Object {
          "end": Object {
            "column": 16,
            "line": 20,
          },
          "start": Object {
            "column": 6,
            "line": 20,
          },
        },
        "mutatorName": "ConditionalExpression",
        "replacement": "true",
        "static": false,
        "status": "Killed",
        "statusReason": "expected true to be false",
        "testsCompleted": 2,
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "14",
        "killedBy": Array [
          "3",
        ],
        "location": Object {
          "end": Object {
            "column": 16,
            "line": 20,
          },
          "start": Object {
            "column": 6,
            "line": 20,
          },
        },
        "mutatorName": "ConditionalExpression",
        "replacement": "false",
        "static": false,
        "status": "Killed",
        "statusReason": "expected false to be true",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "15",
        "killedBy": Array [
          "4",
        ],
        "location": Object {
          "end": Object {
            "column": 16,
            "line": 20,
          },
          "start": Object {
            "column": 6,
            "line": 20,
          },
        },
        "mutatorName": "EqualityOperator",
        "replacement": "number <= 0",
        "static": false,
        "status": "Killed",
        "statusReason": "expected true to be false",
        "testsCompleted": 2,
      },
      Object {
        "coveredBy": Array [
          "3",
          "4",
        ],
        "id": "16",
        "killedBy": Array [
          "3",
        ],
        "location": Object {
          "end": Object {
            "column": 16,
            "line": 20,
          },
          "start": Object {
            "column": 6,
            "line": 20,
          },
        },
        "mutatorName": "EqualityOperator",
        "replacement": "number >= 0",
        "static": false,
        "status": "Killed",
        "statusReason": "expected false to be true",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "3",
        ],
        "id": "17",
        "killedBy": Array [
          "3",
        ],
        "location": Object {
          "end": Object {
            "column": 4,
            "line": 22,
          },
          "start": Object {
            "column": 17,
            "line": 20,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected false to be true",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "3",
        ],
        "id": "18",
        "killedBy": Array [
          "3",
        ],
        "location": Object {
          "end": Object {
            "column": 22,
            "line": 21,
          },
          "start": Object {
            "column": 18,
            "line": 21,
          },
        },
        "mutatorName": "BooleanLiteral",
        "replacement": "false",
        "static": false,
        "status": "Killed",
        "statusReason": "expected false to be true",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "1",
        ],
        "id": "2",
        "killedBy": Array [
          "1",
        ],
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 8,
          },
          "start": Object {
            "column": 42,
            "line": 5,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected undefined to equal 3",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "1",
        ],
        "id": "3",
        "killedBy": Array [
          "1",
        ],
        "location": Object {
          "end": Object {
            "column": 11,
            "line": 6,
          },
          "start": Object {
            "column": 3,
            "line": 6,
          },
        },
        "mutatorName": "UpdateOperator",
        "replacement": "number--",
        "static": false,
        "status": "Killed",
        "statusReason": "expected 1 to equal 3",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "2",
        ],
        "id": "4",
        "killedBy": Array [
          "2",
        ],
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 12,
          },
          "start": Object {
            "column": 42,
            "line": 10,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected undefined to equal -2",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "2",
        ],
        "id": "5",
        "killedBy": Array [
          "2",
        ],
        "location": Object {
          "end": Object {
            "column": 17,
            "line": 11,
          },
          "start": Object {
            "column": 10,
            "line": 11,
          },
        },
        "mutatorName": "UnaryOperator",
        "replacement": "+number",
        "static": false,
        "status": "Killed",
        "statusReason": "expected 2 to equal -2",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [],
        "id": "6",
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 16,
          },
          "start": Object {
            "column": 46,
            "line": 14,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [],
        "id": "7",
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 15,
          },
          "start": Object {
            "column": 10,
            "line": 15,
          },
        },
        "mutatorName": "ConditionalExpression",
        "replacement": "true",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [],
        "id": "8",
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 15,
          },
          "start": Object {
            "column": 10,
            "line": 15,
          },
        },
        "mutatorName": "ConditionalExpression",
        "replacement": "false",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [],
        "id": "9",
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 15,
          },
          "start": Object {
            "column": 10,
            "line": 15,
          },
        },
        "mutatorName": "EqualityOperator",
        "replacement": "number >= 10",
        "static": false,
        "status": "NoCoverage",
      },
    ],
    "source": "module.exports.add = function(num1, num2) {
  return num1 + num2;
};

module.exports.addOne = function(number) {
  number++;
  return number;
};

module.exports.negate = function(number) {
  return -number;
};

module.exports.notCovered = function(number) {
  return number > 10;
};

module.exports.isNegativeNumber = function(number) {
  var isNegative = false;
  if(number < 0){
    isNegative = true;
  }
  return isNegative;
};


",
  },
  "src/Circle.js": Object {
    "language": "javascript",
    "mutants": Array [
      Object {
        "coveredBy": Array [
          "5",
        ],
        "id": "19",
        "killedBy": Array [
          "5",
        ],
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 4,
          },
          "start": Object {
            "column": 52,
            "line": 1,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "Killed",
        "statusReason": "expected undefined to equal 6.283185307179586",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "5",
        ],
        "id": "20",
        "location": Object {
          "end": Object {
            "column": 30,
            "line": 3,
          },
          "start": Object {
            "column": 10,
            "line": 3,
          },
        },
        "mutatorName": "ArithmeticOperator",
        "replacement": "2 * Math.PI / radius",
        "static": false,
        "status": "Survived",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [
          "5",
        ],
        "id": "21",
        "killedBy": Array [
          "5",
        ],
        "location": Object {
          "end": Object {
            "column": 21,
            "line": 3,
          },
          "start": Object {
            "column": 10,
            "line": 3,
          },
        },
        "mutatorName": "ArithmeticOperator",
        "replacement": "2 / Math.PI",
        "static": false,
        "status": "Killed",
        "statusReason": "expected 0.6366197723675814 to equal 6.283185307179586",
        "testsCompleted": 1,
      },
      Object {
        "coveredBy": Array [],
        "id": "22",
        "location": Object {
          "end": Object {
            "column": 2,
            "line": 8,
          },
          "start": Object {
            "column": 46,
            "line": 6,
          },
        },
        "mutatorName": "BlockStatement",
        "replacement": "{}",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [],
        "id": "23",
        "location": Object {
          "end": Object {
            "column": 20,
            "line": 7,
          },
          "start": Object {
            "column": 11,
            "line": 7,
          },
        },
        "mutatorName": "ArithmeticOperator",
        "replacement": "5 / 2 / 3",
        "static": false,
        "status": "NoCoverage",
      },
      Object {
        "coveredBy": Array [],
        "id": "24",
        "location": Object {
          "end": Object {
            "column": 16,
            "line": 7,
          },
          "start": Object {
            "column": 11,
            "line": 7,
          },
        },
        "mutatorName": "ArithmeticOperator",
        "replacement": "5 * 2",
        "static": false,
        "status": "NoCoverage",
      },
    ],
    "source": "module.exports.getCircumference = function(radius) {
  //Function to test multiple math mutations in a single function.
  return 2 * Math.PI * radius;
};

module.exports.untestedFunction = function() {
  var i = 5 / 2 * 3;
};
",
  },
}
`;

exports[`Verify stryker has ran correctly should report json report with expected results 2`] = `
Object {
  "test/unit/AddSpec.js": Object {
    "source": "var addModule = require('../../src/Add');
var add = addModule.add;
var addOne = addModule.addOne;
var isNegativeNumber = addModule.isNegativeNumber;
var negate = addModule.negate;
var notCovered = addModule.notCovered;

describe('Add', function() {
  it('should be able to add two numbers', function() {
    var num1 = 2;
    var num2 = 5;
    var expected = num1 + num2;

    var actual = add(num1, num2);

    expect(actual).to.be.equal(expected);
  });

  it('should be able 1 to a number', function() {
    var number = 2;
    var expected = 3;

    var actual = addOne(number);

    expect(actual).to.be.equal(expected);
  });

  it('should be able negate a number', function() {
    var number = 2;
    var expected = -2;

    var actual = negate(number);

    expect(actual).to.be.equal(expected);
  });

  it('should be able to recognize a negative number', function() {
    var number = -2;

    var isNegative = isNegativeNumber(number);

    expect(isNegative).to.be.true;
  });

  it('should be able to recognize that 0 is not a negative number', function() {
    var number = 0;

    var isNegative = isNegativeNumber(number);

    expect(isNegative).to.be.false;
  });
});
",
    "tests": Array [
      Object {
        "id": "0",
        "name": "Add should be able to add two numbers",
      },
      Object {
        "id": "1",
        "name": "Add should be able 1 to a number",
      },
      Object {
        "id": "2",
        "name": "Add should be able negate a number",
      },
      Object {
        "id": "3",
        "name": "Add should be able to recognize a negative number",
      },
      Object {
        "id": "4",
        "name": "Add should be able to recognize that 0 is not a negative number",
      },
    ],
  },
  "test/unit/CircleSpec.js": Object {
    "source": "var circleModule = require('../../src/Circle');
var getCircumference = circleModule.getCircumference;

describe('Circle', function() {
  it('should have a circumference of 2PI when the radius is 1', function() {
    var radius = 1;
    var expectedCircumference = 2 * Math.PI;

    var circumference = getCircumference(radius);

    expect(circumference).to.be.equal(expectedCircumference);
  });
});
",
    "tests": Array [
      Object {
        "id": "5",
        "name": "Circle should have a circumference of 2PI when the radius is 1",
      },
    ],
  },
}
`;
